application/src/main/java/org/togetherjava/tjbot/commands/reminder/RemindCommand.java

```diff
@@ -0,0 +1,152 @@
+package org.togetherjava.tjbot.commands.reminder;
+
+import net.dv8tion.jda.api.entities.ISnowflake;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.Interaction;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import org.jetbrains.annotations.NotNull;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+import org.togetherjava.tjbot.db.Database;
+
+import java.time.*;
+import java.time.temporal.TemporalAmount;
+import java.util.List;
+
+import static org.togetherjava.tjbot.db.generated.Tables.PENDING_REMINDERS;
+
+/**
+ * Implements the '/remind' command which can be used to automatically send reminders to oneself at
+ * a future date.
+ * <p>
+ * Example usage:
+ *
+ * <pre>
+ * {@code
+ * /remind amount: 5 unit: weeks content: Hello World!
+ * }
+ * </pre>
+ * <p>
+ * Pending reminders are processed and send by {@link RemindRoutine}.
+ */
+public final class RemindCommand extends SlashCommandAdapter {
+    private static final String COMMAND_NAME = "remind";
+    private static final String WHEN_AMOUNT_OPTION = "in-amount";
+    private static final String WHEN_UNIT_OPTION = "in-unit";
+    private static final String CONTENT_OPTION = "content";
+
+    private static final int MIN_WHEN_AMOUNT = 1;
+    private static final int MAX_WHEN_AMOUNT = 1_000;
+    private static final List<String> WHEN_UNITS =
+            List.of("minutes", "hours", "days", "weeks", "months", "years");
+    private static final Period MAX_WHEN_PERIOD = Period.ofYears(3);
+    private static final int MAX_PENDING_REMINDERS_PER_USER = 100;
+
+    private final Database database;
+
+    /**
+     * Creates an instance of the command.
+     *
+     * @param database to store and fetch the reminders from
+     */
+    public RemindCommand(@NotNull Database database) {
+        super(COMMAND_NAME, "Reminds you after a given time period has passed (e.g. in 5 weeks)",
+                SlashCommandVisibility.GUILD);
+
+        // TODO As soon as JDA offers date/time selector input, this should also offer
+        // "/remind at" next to "/remind in" and use subcommands then
+        OptionData whenAmount = new OptionData(OptionType.INTEGER, WHEN_AMOUNT_OPTION,
+                "when to remind you, the amount of the time period (e.g. [5] weeks)", true)
+                    .setRequiredRange(MIN_WHEN_AMOUNT, MAX_WHEN_AMOUNT);
+        OptionData whenUnit = new OptionData(OptionType.STRING, WHEN_UNIT_OPTION,
+                "when to remind you, the unit of the time period (e.g. 5 [weeks])", true);
+        WHEN_UNITS.forEach(unit -> whenUnit.addChoice(unit, unit));
+
+        getData().addOptions(whenAmount, whenUnit)
```

Comment:
I'd personally do unit before amount. People might get confused and think it's the amount in weeks (because of the example)

Reply:
I get that. But it reads kinda weird then `"remind in weeks 5 content huhu"` vs `"remind in 5 weeks content huhu"`.

But yeah, I guess its better like that if we split the input into two sections.

Reply:
done